src/core/providers/gdal/qgsgdalproviderbase.cpp

```diff
@@ -451,13 +477,39 @@ QVariantMap QgsGdalProviderBase::decodeGdalUri( const QString &uri )
         break;
       }
     }
+
+    const thread_local QRegularExpression credentialOptionRegex( QStringLiteral( "\\|credential:([^|]*)" ) );
+    const thread_local QRegularExpression credentialOptionKeyValueRegex( QStringLiteral( "(.*?)=(.*)" ) );
+    while ( true )
+    {
+      const QRegularExpressionMatch match = credentialOptionRegex.match( path );
+      if ( match.hasMatch() )
+      {
+        const QRegularExpressionMatch keyValueMatch = credentialOptionKeyValueRegex.match( match.captured( 1 ) );
+        if ( keyValueMatch.hasMatch() )
+        {
+          credentialOptions.insert( keyValueMatch.captured( 1 ), keyValueMatch.captured( 2 ) );
+        }
+        else
+        {
+          credentialOptions.insert( match.captured( 1 ), QString() );
```

Comment:
I'm not completely sure if we need to handle a credential that is a key without value. I can't think of a case where it is needed currently. Unless this is meant to remove a path specific option. But in that case VSISetPathSpecificOption() should be called with a nullptr value and not an empty string

Reply:
Ok, I'll remove this support. (I wasn't sure if in future there'd be valueless options added in gdal)